MobileBuy/buy/src/main/java/com/shopify/buy/utils/AndroidPayHelper.java

```diff
@@ -0,0 +1,391 @@
+
+        if (buyClient == null) {
+            throw new NullPointerException("buyClient cannot be null");
+        }
+
+        if (apiClient == null) {
+            throw new NullPointerException("apiClient cannot be null");
+        }
+
+        if (delegate == null) {
+            throw new NullPointerException("delegate cannot be null");
+        }
+
+        // Check to see if Google play is up to date and the buyclient has android pay enabled
+        if (!hasGooglePlayServices(context) || !buyClient.androidPayIsEnabled()) {
+            delegate.onResult(false);
+            return;
+        }
+
+        isReadyToPay(apiClient, delegate);
+    }
+
+    /**
+     * Checks to see if Play Services are available on device
+     *
+     * @param context
+     * @return true if Play Services are available
+     */
+    public static boolean hasGooglePlayServices(final Context context) {
+        if (context == null) {
+            throw new NullPointerException("context cannot be null");
+        }
+
+        GoogleApiAvailability googleAPI = GoogleApiAvailability.getInstance();
+        int result = googleAPI.isGooglePlayServicesAvailable(context);
+
+        if (result != ConnectionResult.SUCCESS) {
+            return false;
+        }
+
+        return true;
+    }
+
+
+    /**
+     * Checks to see if the Android Pay App is installed on device and has a valid card for In App Purchase using {@link com.google.android.gms.wallet.Payments#isReadyToPay(GoogleApiClient)}
+     *
+     * @param apiClient The {@link GoogleApiClient}, not null
+     * @param delegate  The {@link AndroidPayAvailableResponse} delegate for receiving the result
+     */
+    public static void isReadyToPay(final GoogleApiClient apiClient, final AndroidPayAvailableResponse delegate) {
+        if (apiClient == null) {
+            throw new NullPointerException("apiClient cannot be null");
+        }
+
+        if (delegate == null) {
+            throw new NullPointerException("delegate cannot be null");
+        }
+
+        // Check that the user has installed and setup the Android Pay app on their device
+        Wallet.Payments.isReadyToPay(apiClient).setResultCallback(
+                new ResultCallback<BooleanResult>() {
+                    @Override
+                    public void onResult(@NonNull BooleanResult booleanResult) {
+
+                        if (booleanResult.getStatus().isSuccess()) {
+                            delegate.onResult(booleanResult.getValue());
+                        } else {
+                            // We could not make the call so must assume it is not available
+                            delegate.onResult(false);
+                        }
+                    }
+                });
+    }
+
+
+    /**
+     * Interface for receiving results from {@link AndroidPayHelper#androidPayIsAvailable(Context, BuyClient, GoogleApiClient, AndroidPayAvailableResponse)}
+     */
+    public interface AndroidPayAvailableResponse {
```

Comment:
Shouldn't it be named like AndroidPayReadyCallback? or similar with *Callback suffix or Listener etc